### DISCUSSION TITLE
Make glossary

### DISCUSSION THREAD

**Discussion by Marsevil** [tectonic-typesetting/tectonic#1063] [Q&A]
Hi everyone,

I everyone,
Today I'm trying to make a glossary in my Latex document. And unfortunately, it does not work.
I found [this issue](https://github.com/tectonic-typesetting/tectonic/issues/115) which sounds related to that but still no contribution at that time.
I'm wondering if somebody found a work around, I tried to run Tectonic with `--keep-intermediates` then run `makeglossaries` manually and, to finish, re-run tectonic, but my glossary still does not show up in the generated document.
Please note that there is no error reported.

**Comment by kpym** [comment]
@Marsevil For me it works. 

Consider the following (`test_glossary.tex`) MWE
```latex
\documentclass[11pt]{article}
\usepackage{fontspec}
\usepackage{glossaries}

\makeglossaries

\newglossaryentry{latex}
{
    name=latex,
    description={Is a markup language specially suited
    for scientific documents}
}

\newglossaryentry{maths}
{
    name=mathematics,
    description={Mathematics is what mathematicians do}
}

\title{How to create a glossary}
\author{ }
\date{ }

\begin{document}
\maketitle

The \Gls{latex} typesetting markup language is specially suitable
for documents that include \gls{maths}.

\clearpage

\printglossaries

\end{document}

```

Then 

```bash
tectonic -k test_glossary.tex
makeglossaries test_glossary
tectonic -k test_glossary.tex
```

produce a pdf with glossary.

**Comment by Marsevil** [comment]
Ok thank you, it works.
In fact, it doesn't work with the new CLI. My guess is that is because sources are not stored in the same place than the build results but I'm not sure.
